src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.domain.Book;
+import mate.academy.lib.Dao;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_BOOK_QUERY = "INSERT INTO book (title, price) VALUES (?, ?)";
+    private static final String FIND_BOOK_BY_ID_QUERY = "SELECT * FROM book WHERE id = ?";
+    private static final String FIND_ALL_BOOK_QUERY = "SELECT * FROM book";
+    private static final String UPDATE_BOOK_QUERY = "UPDATE book SET title = ?, "
+            + "price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_QUERY = "DELETE FROM book WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        Book createdBook;
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement = connection.prepareStatement(CREATE_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(2, book.getTitle());
+            preparedStatement.setBigDecimal(3, book.getPrice());
```

Comment:
The parameters for `preparedStatement.setString` and `preparedStatement.setBigDecimal` should start from index 1, not 2 and 3 respectively. This is a bug that will cause an SQLException due to missing parameter bindings.